src/linker/pipeline_schema.py

```diff
@@ -18,26 +18,28 @@ def get_schemas(cls) -> List["PipelineSchema"]:
         """Creates the allowable schema for the pipeline."""
         schemas = []
         pvs_like_case_study = cls("pvs_like_case_study")
-        pvs_like_case_study.add_step(Step("pvs_like_case_study"))
+        # Add the steps in the order they should be run
+        pvs_like_case_study._add_step(Step("pvs_like_case_study"))
         schemas.append(pvs_like_case_study)
         return schemas
 
-    def add_step(self, step: Step):
+    def _add_step(self, step: Step):
         self.steps.append(step)
 
     def validate_pipeline(self, pipeline: Pipeline):
-        # TODO:
-        ## ensure implementation is assigned to the correct step
-        ## ensure that the implementations are in the correct step order
+        # Loop through the pipeline implementations and check if their order
+        ## and corresponding steps match the schema
+        for idx, implementation in enumerate(pipeline.implementations):
+            if implementation.step != self.steps[idx]:
+                return False
```

Comment:
Rajan - need some implementaton validations. Delete this method here (add it to `pipeline.validate(schema)` and also add:
- do containers exist?
- does each individual implementation match the step that it is implementing?

Reply:
Reading this again, I'm still not totally sure what exactly the second bullet means. Doesn't line 33 above cover this?